manifests/mirror.pp

```diff
@@ -86,6 +98,18 @@
     $components_arg = " ${components}"
   }
 
+  if empty($filter) {
+    $filter_arg = ''
+  } else{
+    $filter_arg = "-filter=\"${filter}\""
```

Comment:
If you add a space before and after the value being assigned to `$filter_arg` and `$filter_with_deps_arg` you can put the values next to each other on line 142. The spec tests for when filter_arg is not present wouldn no longer need to rely on consecutive blank whitespace. Which would also make the change set a fair bit smaller.